packages/mui-material/src/Alert/Alert.d.ts

```diff
@@ -33,6 +35,23 @@ export interface AlertProps extends StandardProps<PaperProps, 'variant'> {
    * [palette customization guide](https://mui.com/material-ui/customization/palette/#adding-new-colors).
    */
   color?: OverridableStringUnion<AlertColor, AlertPropsColorOverrides>;
+  /**
+   * The components used for each slot inside the Alert.
+   * Either a string to use a HTML element or a component.
+   * @default {}
+   */
+  components?: {
+    CloseButton?: React.ElementType;
+    CloseIcon?: React.ElementType;
+  };
+  /**
+   * The props used for each slot inside.
+   * @default {}
+   */
+  componentsProps?: {
+    closeButton?: IconButtonProps;
+    closeIcon?: SvgIconProps;
+  } & AlertComponentsPropsOverrides;
```

Comment:
Let's move forward with this:
```suggestion
  componentsProps?: {
    closeButton?: IconButtonProps;
    closeIcon?: SvgIconProps;
  };
```

Reply:
Agree, let's not introduce new concepts now. I've added this thread to the discussions we need to resolve before the MUI Base goes stable / before Material UI v6 (whichever will be sooner).